File: Python/calculator.py
def calc(num1: float, num2:float, op: str) -> float:
    res: float = 0
    match (op):
        case '-': res = num1 - num2
        case '+': res = num1 + num2
        case '*': res = num1 * num2
        case '/': res = num1 / num2
    return res

num1: float = float(input("Write a: "))
num2: float = float(input("Write b: "))
op: str = input("Write op (+-/*): ")

try:
    answer: float = calc(num1=num1, num2=num2, op=op)
    print(f"Answer {answer}")
except ZeroDivisionError:
    print("ZeroDivisionError")

File: Pascal/calculator.pas
Program calculator;
var num1, num2: real;
var answer: real;
var op: string;

function calc(num1: real; num2: real; op: string):real;
    var res: real;
    begin
      if op='-' then res:=num1-num2
      else 
        if op='+' then res:=num1+num2
      else 
        if op='*' then res:=num1*num2
      else 
        if op='/' then 
          begin
            if num2=0 then 
              begin
                Writeln('ZeroDivisionError');
              end;
            res:=num1/num2;
          end;
      calc:=res;
    end;

begin
  Write('Write a: ');
  readln(num1);
  Write('Write b: ');
  readln(num2);
  Write('Write op: ');
  readln(op);
  answer:=calc(num1, num2, op);
  Writeln('Answer: ', answer);
end.